### PR TITLE
static cost model

### PR DESCRIPTION
**Pull request by huangxu96**
### PR types

New features
### PR changes

Others
### Describe

Add Static CostModel. Static data is based on op benchmark system

### CODE REVIEW

File: python/paddle/cost_model/cost_model.py [hunk 1]
```diff
@@ -53,17 +54,33 @@ def profile_measure(self,
...
-        print("core:<<<<<<<")
 
         cost_model = core.CostModel()
         cost_data = cost_model.ProfileMeasure(device)
-        # cost_list = self.stop_cost_model()
-        # return cost_list
 
+    def static_cost_data(self):
+        static_cost_data_path = os.path.join(
+            os.path.dirname(__file__), "static_op_benchmark.json")
```
Comment (zhhsplendid) [inline python/paddle/cost_model/cost_model.py:63]: A little suggestion, "static_op_benchmark.json" can be set as a constant at head of file. In the code, we may not find it easily. However, you can do it in next PR

File: python/paddle/cost_model/cost_model.py [hunk 2]
```diff
@@ -53,17 +54,33 @@ def profile_measure(self,
...
+        with open(static_cost_data_path, 'r') as load_f:
+            load_dict = json.load(load_f)
```
Comment (zhhsplendid) [inline python/paddle/cost_model/cost_model.py:65]: Just a question, nothing to change: how large is "static_op_benchmark.json" ?

File: python/paddle/cost_model/cost_model.py [hunk 3]
```diff
@@ -53,17 +54,33 @@ def profile_measure(self,
...
+        if op_name == None:
+            raise ValueError(
+                'op_name should not be empty when you want to get static op time'
+            )
 
-cost_model.profile_measure(startup_program, main_program)
+        op_cost = {}
+        for op_data in self._static_cost_data:
+            if (op_data["op"] == op_name) and (dtype in op_data["config"]):
+                if (forward == True):
```
Comment (zhhsplendid) [inline python/paddle/cost_model/cost_model.py]: Clean the code:

`if forward:`

is enough

File: python/paddle/cost_model/cost_model.py [hunk 4]
```diff
@@ -53,17 +54,33 @@ def profile_measure(self,
...
+        static_cost_data_path = os.path.join(
+            os.path.dirname(__file__), "static_op_benchmark.json")
+        with open(static_cost_data_path, 'r') as load_f:
+            load_dict = json.load(load_f)
+        self._static_cost_data = load_dict
+        # return all static cost data
+        return load_dict
 
-cost_model = CostModel()
+    def get_static_op_time(self, op_name, forward=True, dtype="float32"):
```
Comment (zhhsplendid) [inline python/paddle/cost_model/cost_model.py:70]: Please add comment about `forward`, otherwise the parameter is not obvious.

Comment (huangxu96) [reply python/paddle/cost_model/cost_model.py:70]: Done.

File: python/paddle/cost_model/cost_model.py (hunk 3 above)
Comment (huangxu96) [reply python/paddle/cost_model/cost_model.py]: Done.

File: python/paddle/cost_model/cost_model.py (hunk 2 above)
Comment (huangxu96) [reply python/paddle/cost_model/cost_model.py:65]: 550KB.

File: python/paddle/cost_model/cost_model.py (hunk 1 above)
Comment (huangxu96) [reply python/paddle/cost_model/cost_model.py:63]: Cool, will do it in the next PR.

File: python/paddle/cost_model/cost_model.py (hunk 2 above)
Comment (zhhsplendid) [reply python/paddle/cost_model/cost_model.py:65]: OK to me

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Pull request merged**